src/backend/distributed/executor/intermediate_results.c

```diff
@@ -795,8 +795,32 @@ ReadIntermediateResultsIntoFuncOutput(FunctionCallInfo fcinfo, char *copyFormat,
 		int statOK = stat(resultFileName, &fileStat);
 		if (statOK != 0)
 		{
-			ereport(ERROR, (errcode_for_file_access(),
-							errmsg("result \"%s\" does not exist", resultId)));
+			/*
+			 * When the file does not exist, it could mean two different things.
+			 * First -- and a lot more common -- case is that a failure happened
+			 * in a concurrent backend on the same distributed transaction. And,
+			 * one of the backends in that transaction has already been roll
+			 * backed, which has already removed the file. If we throw an error
+			 * here, the user might see this error instead of the actual error
+			 * message. Instead, we prefer to WARN the user and pretend that the
+			 * file has no data in it. In the end, the user would see the actual
+			 * error message for the failure.
+			 *
+			 * Second, in case of any bugs in intermediate result broadcasts,
+			 * we could try to read a non-existing file. That is most likely
+			 * to happen during development. Thus, when asserts enabled, we throw
+			 * an error instead of WARNING so that the developers cannot miss.
+			 */
+			int logLevel = WARNING;
+
+#ifdef USE_ASSERT_CHECKING
+			logLevel = ERROR;
+#endif
+			ereport(logLevel, (errcode_for_file_access(),
```

Comment:
I considered adding a detail saying that  `if you see this and the transaction  doesn't  fail, report a bug `  but that might  be over complicated warning, even if users would see an error afterwards.

Any comments on that?

Reply:
let's not do that

Reply:
This needs a little more work as the regression test outputs change considerable when asserts enabled or not

Reply:
Maybe we always do WARNING? We will typically see it as developers.

Reply:
Applied this feedback,